libs/blocks/email-collection/utils.js

```diff
@@ -0,0 +1,271 @@
+    if (!countryCode) return;
+    formattedData[countryCode] = formattedData[countryCode] ?? [];
+    formattedData[countryCode].push({ key, value });
+  });
+  return formattedData;
+}
+
+function defaultFormatData(data) {
+  return data.reduce((acc, { key, value }) => {
+    acc[key] = value;
+    return acc;
+  }, {});
+}
+
+async function fetchSheet(sheetData) {
+  const formatData = { state: formatStateData };
+  const { id, url } = sheetData;
+  try {
+    const sheetReq = await fetch(url);
+    if (!sheetReq.ok) return { [id]: {} };
+    const { data } = await sheetReq.json();
+
+    if (formatData[id]) return { [id]: formatData[id](data) };
+    return { [id]: defaultFormatData(data) };
+  } catch (e) {
+    window.lana?.log(e);
+    return { [id]: {} };
+  }
+}
+
+async function fetchFormConfig(sheets) {
+  const sheetPromises = [{ url: PLACEHOLDER_URL, id: 'placeholders' }, ...sheets]
+    .map((sheet) => (fetchSheet(sheet)));
+
+  const resolved = await Promise.all(sheetPromises);
+  let config = {};
+  resolved.forEach((result) => { config = { ...config, ...result }; });
+  return config;
+}
+
+export async function fetchConsentString() {
+  try {
+    const stringReq = await fetch(CONSENT_URL);
+    if (!stringReq.ok) return {};
+
+    const string = await stringReq.text();
+    const doc = new DOMParser().parseFromString(string, 'text/html');
+    const [consentDiv, consentId] = doc.querySelectorAll('body > div');
+
+    return {
+      consentDiv,
+      consentId: consentId?.textContent.trim(),
+    };
+  } catch (e) {
+    return {};
+  }
+}
+
+export const [getFormData, setFormData] = (() => {
+  let formData;
+  return [
+    (key) => (formData[key]),
+    (el) => {
+      formData = {
+        fields: {},
+        metadata: {},
+        consent: fetchConsentString(),
+      };
+      const fetchConfigParams = [];
+      const { fields, metadata } = formData;
+      const metadataEl = el.parentElement?.querySelector('.section-metadata');
+      if (!metadataEl) return 'Section metadata is missing';
+
+      [...metadataEl.children].forEach((child) => {
+        const key = formatMetadataKey(child.firstElementChild?.textContent);
+        if (!FORM_FIELDS[key] && !FORM_METADATA[key]) return;
+        const value = child.lastElementChild.textContent;
+        const metadataObject = FORM_METADATA[key] ? metadata : fields;
+        const newKey = FORM_METADATA[key] ?? key;
+        metadataObject[newKey] = value;
+
+        if (FORM_FIELDS[key]?.url) fetchConfigParams.push({ id: key, url: FORM_FIELDS[key].url });
+      });
+
+      if (!fields.email
+        || !metadata.mpsSname
+        || !metadata.subscriptionName) return 'Section metadata is missing email/mps-sname/subscription-name field';
+
+      formData.config = fetchFormConfig(fetchConfigParams);
+
+      return null;
+    },
+  ];
+})();
+
+async function getMartech() {
+  // eslint-disable-next-line
+  const satellite = await window.__satelliteLoadedPromise;
```

Comment:
We need some error handling for this, I think you'll encounter problems when running with `?martech=off`